### PR TITLE
Disable stable ID for contracts

### PR DESCRIPTION
**Pull request by ununhexium**
_What issues does this PR close?_

```[tasklist]
### Checklist
- [x] The PR title is short and expressive.
- [x] I have updated the CHANGELOG.md. See [changelog_update.md](https://github.com/sovity/authority-portal/tree/main/docs/dev/changelog_updates.md) for more information.
- [x] I have updated the Deployment Migration Notes Section in the CHANGELOG.md for any configuration / external API changes.
- [x] I have performed a **self-review**
```

### CODE REVIEW

File: utils/catalog-parser/src/main/java/de/sovity/edc/utils/catalog/mapper/DspDataOfferBuilder.java [hunk 1]
```diff
@@ -22,7 +22,6 @@
 import jakarta.json.Json;
 import jakarta.json.JsonObject;
```
Comment (richardtreier) [inline utils/catalog-parser/src/main/java/de/sovity/edc/utils/catalog/mapper/DspDataOfferBuilder.java:23]: Missing CHANGELOG Entry.

File: utils/catalog-parser/src/test/java/de/sovity/edc/utils/catalog/mapper/DspContractOfferUtilsTest.java [hunk 2]
```diff
@@ -2,11 +2,13 @@
 
 import jakarta.json.Json;
 import lombok.val;
+import org.junit.jupiter.api.Disabled;
 import org.junit.jupiter.api.Test;
 
 import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
 
 class DspContractOfferUtilsTest {
+    @Disabled("hotfix: the stable ID or the way the new ID is build prevents a contract negotiation.")
```
Comment (richardtreier) [inline utils/catalog-parser/src/test/java/de/sovity/edc/utils/catalog/mapper/DspContractOfferUtilsTest.java:11]: I think it's fine to remove this test, if we have / add a dedicated test for the stable id stuff. We could then use the Stable ID Code from the Broker

**Review by richardtreier — CHANGES_REQUESTED** [review]

File: utils/catalog-parser/src/test/java/de/sovity/edc/utils/catalog/mapper/DspContractOfferUtilsTest.java (hunk 2 above)
Comment (ununhexium) [reply utils/catalog-parser/src/test/java/de/sovity/edc/utils/catalog/mapper/DspContractOfferUtilsTest.java:11]: ok

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -6,7 +6,7 @@ on:
   release:
     types: [ published ]
   pull_request:
-    branches: [ main ]
+    branches: [ main, version/v7.2.1 ]
```
Comment (tmberthold) [inline .github/workflows/ci.yml:9]: if you changed the base-branch to main, you dont needto include this one

Comment (ununhexium) [reply .github/workflows/ci.yml]: Reverted to main only.

File: CHANGELOG.md [hunk 4]
```diff
@@ -22,6 +22,39 @@ please see [changelog_updates.md](docs/dev/changelog_updates.md).
...
+
+_No special deployment migration steps required_
+
+#### Compatible Versions
+
+- Connector Backend Docker Images:
+    - Dev EDC: `ghcr.io/sovity/edc-dev:7.2.2`
+    - sovity EDC CE: `ghcr.io/sovity/edc-ce:7.2.2`
+    - MDS EDC CE: `ghcr.io/sovity/edc-ce-mds:7.2.2`
+- Connector UI Docker Image: `ghcr.io/sovity/edc-ui:2.4.0`
```
Comment (tmberthold) [inline CHANGELOG.md:55]: Can we stay on edc-ui 2.4.0 if we merge it into main and do a release from main or do we need to upgrade to the latest v2.5.0? https://github.com/sovity/edc-ui/releases/tag/v2.5.0

Comment (richardtreier) [reply CHANGELOG.md:55]: No need to migrate, as 2.5.0 was only released for the broker

**Comment by tmberthold** [comment]
Related: https://github.com/sovity/PMO-Software/issues/754 (to have the link)

**Pull request merged**